operators/src/engine/query_processor.rs

```diff
@@ -248,6 +267,31 @@ impl TypedVectorQueryProcessor {
     }
 }
 
+/// An enum that contains all possible query processor variants
+#[allow(clippy::pub_enum_variant_names)]
+pub enum TypedPlotQueryProcessor {
```

Comment:
The json type I get but do we really have to commit to PNG here? What if we want to deliver plots as .webp or SVG? Do we introduce more types? Does the client have to care about the image format?

Reply:
We could either go with new types or bytes and an image flag. At some point, however, or service needs to reply with the correct mime type. I guess we can refactor this if we know more about our plot ambitions. PNG and JSON are the mapping responses.